github/pulls.go

```diff
@@ -250,17 +250,20 @@ type PullRequestMergeResult struct {
 
 type pullRequestMergeRequest struct {
 	CommitMessage *string `json:"commit_message"`
+	Squash        *bool   `json:"squash"`
 }
 
 // Merge a pull request (Merge Button™).
 //
 // GitHub API docs: https://developer.github.com/v3/pulls/#merge-a-pull-request-merge-buttontrade
-func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string) (*PullRequestMergeResult, *Response, error) {
+func (s *PullRequestsService) Merge(owner string, repo string, number int, commitMessage string, squash bool) (*PullRequestMergeResult, *Response, error) {
 	u := fmt.Sprintf("repos/%v/%v/pulls/%d/merge", owner, repo, number)
 
 	req, err := s.client.NewRequest("PUT", u, &pullRequestMergeRequest{
 		CommitMessage: &commitMessage,
+		Squash:        &squash,
 	})
+	req.Header.Add("Accept", mediaTypeSquashPreview)
```

Comment:
You probably want to add a `TODO` comment above this line:
`// TODO: remove custom Accept header when this API fully launches.`